### PR TITLE
Trigger error in case of SRDF syntax error

### PR DESCRIPTION
**Pull request by kkufieta**
This is to help with this [issue](https://github.com/ros-planning/moveit/issues/967) in MoveIt!

### CODE REVIEW

**Review by rhaschke — APPROVED** [review]
LGTM. Thanks!
Did you checked that this `false` return value is correctly handled by callers?

**Comment by rhaschke** [comment]
Fixes https://github.com/ros-planning/moveit/issues/967.

File: src/model.cpp [hunk 1]
```diff
@@ -628,6 +628,11 @@ bool srdf::Model::initString(const urdf::ModelInterface &urdf_model, const std::
 {
   TiXmlDocument xml_doc;
   xml_doc.Parse(xmlstring.c_str());
+  if (xml_doc.Error())
+    {
+      CONSOLE_BRIDGE_logError("Could not parse the SRDF XML File. %s", xml_doc.ErrorDesc());
+      return false;
+    }
```
Comment (davetcoleman) [inline src/model.cpp]: indentation bad: https://github.com/ros-planning/srdfdom/pull/42

**Review by davetcoleman — APPROVED** [review]

**Comment by kkufieta** [comment]
@rhaschke `initString` already forwards `false` from `initXml` when it fails to find the `robot` element in the xml file. My change just catches the syntax failure case in addition and treats it the same way.

**Comment by jonbinney** [comment]
@kkufieta can you rebase and run clang-format?

**Review by davetcoleman — APPROVED** [review]

**Pull request merged**